### ISSUE TITLE
const N doesn't overflow

### ISSUE THREAD

**Issue by hjorthjort** [kframework/wasm-semantics#48]
As an invariant, for any integer `< iNN > I:Int` on the stack, `I` **should** be between 0 and `#pow(NN) - 1`. For `const` operations, negative values and too large values may be pushed to the stack.

**Comment by hjorthjort** [comment]
@ehildenb In `#assertTopStack`, we currently use `#unsigned` to convert the argument

```
    rule <k> #assertTopStack < ITYPE:IValType > VAL _ => . ... </k> <stack> < ITYPE > VAL' : STACK => STACK </stack>
      requires #unsigned(ITYPE, VAL) ==Int VAL'
```
I think it would make more sense if `#assertTopStack` checked the stack exactly, to avoid exactly this kind of thing, but I would like to hear your thoughts.

**Comment by hjorthjort** [comment]
Closed by #49.
